src/core/include/openvino/core/type/nf4.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "openvino/core/core_visibility.hpp"
+#include "openvino/core/type/float16.hpp"
+
+#define ROUND_MODE_TO_NEAREST_EVEN
+
+namespace ov {
+class OPENVINO_API ConvertNF4 {
```

Comment:
do we need it in public API?
I see it's used internally only. So, proper place is this folder https://github.com/openvinotoolkit/openvino/tree/0915a150c38fa3c29d08fa99ea0b88ef9776f4d1/src/core/dev_api